### PR TITLE
`sparse_categorical_crossentropy` should check bounds

### PR DESCRIPTION
**Pull request by alexander-g**
`jax.take_along_axis` does not throw an exception when the indices are out of bounds. This can lead to incorrect values in `sparse_categorical_crossentropy` without a warning or error. I have added the `check_bounds` (default: `True`) parameter which sets the loss to `NaN` when there are invalid values in `y_true`.

### CODE REVIEW

**Comment by cgarciae** [comment]
@alexander-g  thanks! This makes a lot of sense, specially since its opt-in.

**Comment by alexander-g** [comment]
@cgarciae No, actually this is **opt-out**, it's on by default
The idea is that you usually don't expect invalid values in your `y_true` so checking bounds by default catches this. Only when you are completely sure that everything is fine you can disable it with `check_bounds=False` to avoid the (tiny) overhead.
